### ISSUE TITLE
Chamfer distance can OOM when there are less than 10k points

### ISSUE THREAD

**Issue by thetianshuhuang** [RadarML/nrdk#28]
When fixing #24, #26 introduces a new bug. Specifically, `_limit` returns `Float[...]` (nominally logits) when callers may be expecting `Bool[...]`. This leads to OOM, as well as potential calculation errors if the output is interpreted as bool via `torch.where` and other checks.
